src/sys/redox/mod.rs

```diff
@@ -0,0 +1,31 @@
+mod awakener;
+mod eventedfd;
+mod io;
+mod selector;
+mod tcp;
+mod udp;
+
+pub use self::awakener::Awakener;
+pub use self::eventedfd::EventedFd;
+pub use self::io::{Io, set_nonblock};
+pub use self::selector::{Events, Selector};
+pub use self::tcp::{TcpStream, TcpListener};
+pub use self::udp::UdpSocket;
+
+use std::os::unix::io::FromRawFd;
+
+pub fn pipe() -> ::io::Result<(Io, Io)> {
```

Comment:
It's unclear to my which `Result` your using here, I assume it's `std::io::Result`, but starting with `::` I'm not sure.

Reply:
Well I would prefer it to be a bit clearer, but that can be left for a future pr then.